server/http.go

```diff
@@ -411,42 +692,67 @@ func (p *Plugin) handleStartMeeting(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	client, _, err := p.getActiveClient(user)
-	if err != nil {
-		p.API.LogWarn("Error getting the client", "err", err)
-		return
+	topic := req.Topic
+	if topic == "" {
+		topic = DefaultMeetingTopic
 	}
 
-	meeting, err := client.CreateMeeting(zoomUser, defaultMeetingTopic)
-	if err != nil {
-		p.API.LogWarn("Error creating the meeting", "err", err)
+	var meetingID int
+	var createMeetingErr error = nil
+	userPMISettingPref, getUserPMISettingErr := p.getPMISettingData(user.Id)
+	if getUserPMISettingErr != nil {
+		p.askUserForMeetingPreference(user.Id, req.ChannelID)
 		return
 	}
 
-	meetingID := meeting.ID
-	if err = p.postMeeting(user, meetingID, req.ChannelID, "", req.Topic); err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+	switch userPMISettingPref {
+	case zoomPMISettingValueAsk:
+		p.askUserForMeetingPreference(user.Id, req.ChannelID)
 		return
+	case "", trueString:
+		meetingID = zoomUser.Pmi
+	default:
+		meetingID, createMeetingErr = p.createMeetingWithoutPMI(user, zoomUser, req.ChannelID, topic)
 	}
 
-	p.trackMeetingStart(userID, telemetryStartSourceWebapp)
+	if meetingID != -1 && createMeetingErr == nil {
+		if err = p.postMeeting(user, meetingID, req.ChannelID, "", topic); err == nil {
+			p.trackMeetingStart(userID, telemetryStartSourceWebapp)
+			if r.URL.Query().Get("force") != "" {
+				p.trackMeetingForced(userID)
+			}
 
-	if r.URL.Query().Get("force") != "" {
-		p.trackMeetingForced(userID)
+			meetingURL := p.getMeetingURL(user, meetingID)
+			if _, err = w.Write([]byte(fmt.Sprintf(`{"meeting_url": "%s"}`, meetingURL))); err != nil {
+				p.API.LogWarn("failed to write the response", "Error", err.Error())
+			}
+			return
+		}
 	}
+	http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
Why do we assume `err` is not `nil` here?

Reply:
If will never be `nil` here, as we are already returning earlier if this `err` is `nil`.